File: learning_tf/src/frame_tf_broadcaster2.cpp
#include <ros/ros.h>
#include <tf/transform_broadcaster.h>

/**
 * フレームを追加するC++のサンプル2
 */

int main(int argc, char** argv){
  // ノード名「my_tf_frame2_broadcaster」
  ros::init(argc, argv, "my_tf_frame2_broadcaster");
  ros::NodeHandle node;

  // Broadcasterとtransform
  tf::TransformBroadcaster br;
  tf::Transform transform;

  ros::Rate rate(10.0); // 10Hz

  // ROSが起動している限り無限ループ
  while (node.ok()){
    // tramsformの設定
    // 時間によって直径2の円周上を移動
    transform.setOrigin( tf::Vector3(2.0*sin(ros::Time::now().toSec()), 2.0*cos(ros::Time::now().toSec()), 0.0) );
    transform.setRotation( tf::Quaternion(0, 0, 0, 1) );

    // turtle1を親とするcarrot1フレームを作成して送信
    // 送信するtramsformの設定にあるように、turtle1を中心とした円周上をcarrot1の基準点が移動
    br.sendTransform(tf::StampedTransform(transform, ros::Time::now(), "turtle1", "carrot1"));
    rate.sleep();
  }

  return 0;
};

File: learning_tf/scripts/frame_tf_broadcaster2.py
#!/usr/bin/python3
import roslib
roslib.load_manifest('learning_tf')

import rospy
import tf
import math

"""
フレームを追加するサンプル2(python)
"""

if __name__ == '__main__':
  # ノード名「my_tf_frame2_broadcaster」
  rospy.init_node('my_tf_frame2_broadcaster')

  # Broadcaster
  br = tf.TransformBroadcaster()

  rate = rospy.Rate(10.0) # 10Hz

  # ROSが起動している限り無限ループ
  while not rospy.is_shutdown():
    # 現在時刻の秒数にPIをかける
    t = rospy.Time.now().to_sec() * math.pi

    # turtle1を親とするcarrot1フレームを作成して送信
    # 時間に応じて円周上をcarrot1の基準点が移動
    br.sendTransform((2.0 * math.sin(t), 2.0 * math.cos(t), 0.0), (0.0, 0.0, 0.0, 1.0), rospy.Time.now(), "carrot1", "turtle1")
    rate.sleep()
